nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -70,28 +71,72 @@ public OAuth2Client getClient(String clientId) {
         return queryClients().stream().map(OAuth2Client::fromDocumentModel).collect(Collectors.toList());
     }
 
+    @Override
+    public OAuth2Client create(OAuth2Client oAuth2Client) {
+        Objects.requireNonNull(oAuth2Client, "oAuth2Client is required");
+        if (getClientModel(oAuth2Client.getId()) != null) {
+            throw new NuxeoException(String.format("Client with id '%s' already exist", oAuth2Client.getId()),
+                    SC_BAD_REQUEST);
+        }
+
+        return execute((session) -> {
+            DocumentModel documentModel = OAuth2Client.fromOAuth2Client(oAuth2Client);
+            return OAuth2Client.fromDocumentModel(session.createEntry(documentModel));
+        });
+    }
+
+    @Override
+    public OAuth2Client update(String clientId, OAuth2Client oAuth2Client) {
+        DocumentModel doc = getClientModel(clientId);
+        if (doc == null) {
+            throw new NuxeoException(SC_NOT_FOUND);
+        }
+
+        return execute((session) -> {
+            OAuth2Client.updateDocument(doc, oAuth2Client);
+            session.updateEntry(doc);
```

Comment:
Maybe clearer to have:
```
DocumentModel updatedDocument = OAuth2Client.updateDocument(doc, oAuth2Client);
session.updateEntry(updatedDocument);
```
since `updateDocument` returns a `DocumentModel`.